src/mumble/Log.cpp

```diff
@@ -333,35 +333,38 @@ void Log::clearIgnore() {
 	qmIgnore.clear();
 }
 
+/// Estimate the size of buf when encoded as an HTML image tag
+/// with the src attribute set to a base64-encoded data URL, such
+/// as:
+///
+///    <img src="image/jpeg;base64,[base64-of-buf-goes-here]" />
+///
+static unsigned int estimatedDataURLImageSize(const QByteArray &buf) {
+    // We estimate the extra characters (HTML) in the encoded
+    // message to be around 64.
+    const unsigned int htmlExtras = 64U;
```

Comment:
If we calculate the exact size required, there is the possibility that the user will have no more space to type a message after inserting the image. Of course, the current value of `htmlExtras` only gives them a few more characters anyway.

In an ideal packing function, the user would insert all of the images/text that he wants into the text message first, then when he hits send, the images are compressed as necessary.